### ISSUE TITLE
expose `World3D.compositor`

### ISSUE THREAD

**Issue by nonchip** [godotengine/godot-proposals#9815]
### Describe the project you are working on

a script (planned addon) to manage WorldEnvironment and Camera related effects

### Describe the problem or limitation you are having in your project

`World3D` exposes `environment` and `camera_attributes` and does have but does not expose the `compositor` property.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

it would allow accessing the `compositor` property right on the `World3D`, where it "matters", instead of having to go through a "dummy" WorldEnvironment node's internal setter or the Server.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

by being the `World3D.compositor` it already is, but visible through the ClassDB

### If this enhancement will not be used often, can it be worked around with a few lines of script?

it can be worked around by adding a WorldEnvironment node and setting the property on it instead.

### Is there a reason why this should be core and not an add-on in the asset library?

because that's where it already is, we just miss 1 line of telling the ClassDB about it.

**Comment by fire** [comment]
> World3D exposes environment and camera_attributes and does have but does not expose the compositor property.

> A script (planned addon) to manage WorldEnvironment and Camera related effects

This is not a game problem. What is the problem you are solving in your game?

**Comment by nonchip** [comment]
@fire i have not the slightest clue by what your example is about, but that *really* doesn't sound related to `World3D` and its existing `compositor` property.

and the problem is that i want to access `World3D.compositor` without abusing a `WorldEnvironment` node. and that the implementor(s) of `World3D.compositor`/`Compositor`/`CompositorEffect` forgot to expose the property to the ClassDB.

**Comment by fire** [comment]
I have no clue why you would want to use World3D and its compositor property so I'm trying to figure out.

**Comment by nonchip** [comment]
@fire yeah i wasnt aware this issue is supposed to document existing features just for mentioning them. usually people look things up in docs or ask if they don't know what a builtin class does.

and get/set "environment" are there, as explained twice, just not "compositor".

**Comment by Zylann** [comment]
I work on a plugin that is pretty much a node in the scene doing stuff, but some of that stuff requires inserting an effect to the compositor, and that complicates things unnecessarily.
My options:
- Ask the user to manually setup the right CompositorEffect and somehow link it to a property of my node. I like when stuff has nice working defaults out of the box, and this is not it.
- Make my node search the entire scene tree for a `WorldEnvironment` until I can find the compositor and finally automatically add a compositor effect there. This probably works but is unnecessarily complicated.
- Just access `get_world_3d().compositor`. But that one is precisely not exposed.

Note: reasons I have a node separate from the effect:
- The node defines a place and shape in the scene that is easier to setup
- The effect doesnt have access to the scene tree AFAIK
